cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -288,8 +288,21 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 	a.DebuggingSnapshotter.StartDataCollection()
 	defer a.DebuggingSnapshotter.Flush()
 
-	unschedulablePodLister := a.UnschedulablePodLister()
-	scheduledPodLister := a.ScheduledPodLister()
+	scheduledAndUnscheduledPodLister := a.ScheduledAndUnscheduledPodLister()
+	scheduledAndUnscheduledPods, err := scheduledAndUnscheduledPodLister.List()
+	if err != nil {
+		klog.Errorf("Failed to list scheduled and unscheduled pods: %v", err)
+		return caerrors.ToAutoscalerError(caerrors.ApiCallError, err)
+	}
+
+	var originalScheduledPods, unschedulablePods []*apiv1.Pod
```

Comment:
nit: let's extract this logic to a separate funcion

Reply:
Done